services/things/persistence/src/main/java/org/eclipse/ditto/services/things/persistence/actors/strategies/events/AbstractThingEventStrategy.java

```diff
@@ -46,12 +51,23 @@ protected AbstractThingEventStrategy() {
     @Override
     public Thing handle(final T event, @Nullable final Thing thing, final long revision) {
         if (null != thing) {
-            final String issuedAt = event.getDittoHeaders().get("_issuedAt");
+            // Create Metadata from Header, see https://github.com/eclipse/ditto/issues/680#issuecomment-654165747
+            // Use existing Metadata or create a new Builder if none exists
+            MetadataBuilder metadataBuilder = thing.getMetadata()
+                .map(Metadata::toBuilder)
+                .orElse(Metadata.newBuilder());
+            for (String headerKey : event.getDittoHeaders().keySet()) {
+                if (headerKey.startsWith(DITTO_HEADER_METADATA_PREFIX)) {
+                    String metadataKey = headerKey.substring(DITTO_HEADER_METADATA_PREFIX.length());
+                    String metadataValue = event.getDittoHeaders().get(headerKey);
+                    metadataBuilder.set(JsonPointer.of(metadataKey), metadataValue);
+                }
+            }
```

Comment:
I'm afraid that it's not that easy.
In https://github.com/eclipse/ditto/issues/680#issuecomment-654165747 I mentioned different syntaxes for the `ditto-metadata:`:
* "ditto-metadata:issuedAt": "2020-06-09T14:29:00Z"
   * the idea of this syntax is that each JSON "leaf" which was updated in the `event` shall receive the metadata `"issuedAt": "2020-06-09T14:29:00Z"`
   * therefore, the `event.getResourcePath()` has to be taken into account (as base path of which part of the thing was changed - this will be `/` when the complete thing was modified) , combined with all pathes of the JSON leafes which can be found out via `event.getEntity(event.getImplementedSchemaVersion())`
   * a JsonObject has to be created for each JSON leaf containing the set metadata
* "ditto-metadata:/features/lamp/properties/color/r/issuedAt": "2020-06-09T14:14:00Z"
   * the idea of this syntax is to specify "overwrites" for the syntax above

For example:
You modify the feature properties of a "Lamp" feature in this Thing:
```json
{
  "thingId": "org.eclipse.ditto:my-lamp",
  "policyId": "org.eclipse.ditto:my-policy",
  "features": {
    "Lamp": { 
      "properties": {
        "on": true,
        "dimmer": 0.8
      }
    }
  }
}
```

You modify `"/features/Lamp/properties"` to:
```json
{
  "on": false,
  "dimmer": 0.0
}
```

The event looks like:
* `getResourcePath()` returns a JsonPointer with `"/features/Lamp/properties"`
*  `event.getEntity(event.getImplementedSchemaVersion())` returns the above JsonObject

If now the header `"ditto-metadata:issuedAt": "2020-06-09T14:29:00Z"` was set, we want to have the following thing JSON (incl. `_metadata`) as a result:
```json
{
  "thingId": "org.eclipse.ditto:my-lamp",
  "policyId": "org.eclipse.ditto:my-policy",
  "features": {
    "Lamp": { 
      "properties": {
        "on": false,
        "dimmer": 0.0
      }
    }
  },
  "_metadata": {
    "features": {
      "Lamp": { 
        "properties": {
          "on": {
            "issuedAt": "2020-06-09T14:29:00Z"
          },
          "dimmer": {
            "issuedAt": "2020-06-09T14:29:00Z"
          }
        }
      }
    }
  }
}
```

So the mechanism is a lot more complex and I think we should extract it to another class in order to be able to also better unit test it.

Ah, and the idea would be to **always** add the following metadata for all affected JSON leafes:
* `_revision`: is passed in via the parameter `revision`
* `_modified`: can be obtained by `event.getTimestamp().orElse(null)`

Reply:
Thanks for the pointers. Indeed it was not meant as complete implementation but a first draft. But your comment helps me. I will try to extract that to another class and add some tests to get a better understanding :)